src/components/PostsList/PostsList.tsx

```diff
@@ -1,37 +1,55 @@
 import React from 'react';
+import { Post } from '../../types/Post';
+import { SelectedPostId } from '../../types/SelectedPostId';
+
 import './PostsList.scss';
 
-export const PostsList: React.FC = () => (
+type Props = {
+  posts: Post[]
+  selectedPostId: SelectedPostId
+  setSelectedPostId: React.Dispatch<React.SetStateAction<SelectedPostId>>
+};
+
+export const PostsList: React.FC<Props> = (
+  {
+    posts,
+    selectedPostId,
+    setSelectedPostId,
+  },
+) => (
   <div className="PostsList">
     <h2>Posts:</h2>
 
-    <ul className="PostsList__list">
-      <li className="PostsList__item">
-        <div>
-          <b>[User #1]: </b>
-          sunt aut facere repellat provident occaecati excepturi optio
-        </div>
-        <button
-          type="button"
-          className="PostsList__button button"
-        >
-          Close
-        </button>
-      </li>
-
-      <li className="PostsList__item">
-        <div>
-          <b>[User #2]: </b>
-          et ea vero quia laudantium autem
-        </div>
-
-        <button
-          type="button"
-          className="PostsList__button button"
-        >
-          Open
-        </button>
-      </li>
-    </ul>
+    {posts.length === 0
+      ? <h3>No posts from this user yet</h3>
+      : (
+        <ul className="PostsList__list">
+          {posts.map(post => (
+            <li className="PostsList__item" key={post.id}>
+              <div>
+                <b>{`[User #${post.userId}]: `}</b>
+                {post.title}
+              </div>
+              {post.id === selectedPostId ? (
+                <button
+                  type="button"
+                  className="PostsList__button button"
+                  onClick={() => setSelectedPostId(null)}
+                >
+                  Close
+                </button>
+              ) : (
+                <button
+                  type="button"
+                  className="PostsList__button button"
+                  onClick={() => setSelectedPostId(post.id)}
+                >
+                  Open
+                </button>
```

Comment:
DRY
you can check only what to pass in the callback and what text to render. No need to repeat 2 similar buttons